NETCORE/test/IntegrationTests.Tests/RequestCollectionTest.cs

```diff
@@ -30,6 +30,101 @@ public RequestCollectionTest(CustomWebApplicationFactory<Startup> factory, ITest
+        {
+            // Arrange
+            var client = _factory.CreateClient();
+            var path = "Home/Privacy";
+            var url = client.BaseAddress + path;
+
+            // Act
+            var request = CreateRequestMessage();
+            request.RequestUri = new Uri(url);
+            var response = await client.SendAsync(request);
+
+            // Assert
+            response.EnsureSuccessStatusCode();
+
+            WaitForTelemetryToArrive();
+
+            var items = _factory.sentItems;
+            PrintItems(items);
+            Assert.Equal(1, items.Count);
+
+            var reqs = GetTelemetryOfType<RequestTelemetry>(items);
+            Assert.Single(reqs);
+            var req = reqs[0];
+            Assert.NotNull(req);
+            ValidateRequest(req, "200", "GET " + path, url, true);
+        }
+
+        [Fact]
+        public async Task RequestFailed()
+        {
+            // Arrange
+            var client = _factory.CreateClient();
+            var path = "Home/Error";
+            var url = client.BaseAddress + path;
+
+            // Act
+            var request = CreateRequestMessage();
+            request.RequestUri = new Uri(url);
+            var response = await client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.InternalServerError, response.StatusCode);
+
+            WaitForTelemetryToArrive();
+
+            var items = _factory.sentItems;
+            PrintItems(items);
+            Assert.Equal(2, items.Count);
+
+            var reqs = GetTelemetryOfType<RequestTelemetry>(items);
+            var exceptions = GetTelemetryOfType<ExceptionTelemetry>(items);
+            Assert.Single(reqs);
+            Assert.Single(exceptions);
+
+            var req = reqs[0];
+            var exc = exceptions[0];
+            Assert.NotNull(req);
+            Assert.NotNull(exc);
+
+            Assert.Equal(exc.Context.Operation.Id, req.Context.Operation.Id);
+            ValidateRequest(req, "500", "GET " + path, url, false);
+        }
+
+        [Fact]
+        public async Task RequestNonExistentPage()
+        {
+            // Arrange
+            var client = _factory.CreateClient();
+            var path = "Home/Nonexistent";
+            var url = client.BaseAddress + path;
+
+            // Act
+            var request = CreateRequestMessage();
+            request.RequestUri = new Uri(url);
+            var response = await client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+
+            WaitForTelemetryToArrive();
+
+            var items = _factory.sentItems;
+            PrintItems(items);
+            Assert.Equal(1, items.Count);
+
+            var reqs = GetTelemetryOfType<RequestTelemetry>(items);
+            Assert.Single(reqs);
+            var req = reqs[0];
+            Assert.NotNull(req);
+
+            ValidateRequest(req, "404", "GET /" + path, url, false);
```

Comment:
nit: I think the tests would be more readable if your call ValidateRequest using the parameter names.

```suggestion
            ValidateRequest(
                 requestTelemetry: req,
                 expectedResponseCode: "404",
                 expectedName: "GET /" + path,
                 expectedUrl: url,
                 expectedSuccess: false);
```

other than this, LGTM :)